transformation/src/main/scala/org/broadinstitute/monster/hca/HcaPipelineBuilder.scala

```diff
@@ -173,33 +166,31 @@ object HcaPipelineBuilder extends PipelineBuilder[Args] {
     val (linksId, linksVersion, projectId) = getLinksIdVersionAndProjectId(fileName)
     // put values in the form we want
     Obj(
-      mutable.LinkedHashMap[Msg, Msg](
-        Str("content") -> Str(encode(metadata)),
-        Str("links_id") -> Str(linksId),
-        Str("version") -> Str(linksVersion),
-        Str("project_id") -> Str(projectId)
-      )
+      Str("content") -> Str(encode(metadata)),
+      Str("links_id") -> Str(linksId),
+      Str("version") -> Str(linksVersion),
+      Str("project_id") -> Str(projectId)
     )
   }
 
   /**
-    * Extract the necessary info from file metadata and put it into a form that
-    * can be used to generate bulk file ingest requests.
-    * @param metadata the content of the metadata file in Msg format
+    * Convert a file descriptor into a row in a bulk-file ingest request for the TDR.
+    *
+    * @param descriptor the content of a descriptor JSON file in Msg format
     * @param inputPrefix the root directory containing input files
-    * @return a Msg object in the desired output format
     */
-  def generateFileIngestRequest(metadata: Msg, inputPrefix: String): Msg = {
-    val contentHash = metadata.read[String]("crc32c")
-    val targetPath = metadata.read[String]("virtual_path")
-    val sourcePath = s"$inputPrefix/data$targetPath"
+  def generateFileIngestRequest(
+    descriptor: Msg,
+    entityType: String,
+    inputPrefix: String
+  ): (String, Msg) = {
+    val contentHash = descriptor.read[String]("crc32c")
+    val targetPath = descriptor.read[String]("file_name")
+    val sourcePath = s"$inputPrefix/data/$targetPath"
 
-    Obj(
-      mutable.LinkedHashMap[Msg, Msg](
-        Str("source_path") -> Str(sourcePath),
-        Str("target_path") -> Str(targetPath),
-        Str("crc32c") -> Str(contentHash)
```

Comment:
did you intend to drop the crc32c?

Reply:
oh, actually, I think that was moved to the line above (187)
`val contentHash = descriptor.read[String]("crc32c")`

Reply:
Yes. It's still in the descriptor JSON, and we can extract it using BigQuery's JSON functions.